pkg/server/test/property_value_test.go

```diff
@@ -44,36 +59,82 @@ func Test_getPropertyValueHappyCase(t *testing.T) {
 				RefID:     "A",
 				QueryType: models.QueryTypePropertyValue,
 				TimeRange: timeRange,
-				JSON: testdata.SerializeStruct(t, models.AssetPropertyValueQuery{
-					BaseQuery: models.BaseQuery{
-						AwsRegion:  testdata.AwsRegion,
-						AssetId:    testdata.DemoTurbineAsset1,
-						PropertyId: testdata.TurbinePropWindSpeed,
-					},
-				}),
+				JSON: []byte(
+					`{
+					   "region":"us-west-2",
+					   "assetId":"1assetid-aaaa-2222-bbbb-3333cccc4444",
+						 "propertyId":"11propid-aaaa-2222-bbbb-3333cccc4444"
+					}`),
 			},
 		},
 	})
 	require.Nil(t, err)
+	_, ok := qdr.Responses["A"]
+	require.True(t, ok)
+	require.NotNil(t, qdr.Responses["A"].Frames[0])
 
-	for i, dr := range qdr.Responses {
-		fname := fmt.Sprintf("%s-%s.golden", "property-value", i)
-		experimental.CheckGoldenJSONResponse(t, "../../testdata", fname, &dr, true)
+	expectedFrame := data.NewFrame("Demo Turbine Asset 1",
+		data.NewField("time", nil, []time.Time{time.Date(2021, 2, 1, 19, 20, 0, 0, time.UTC)}),
+		data.NewField("Wind Speed", nil, []float64{23.8}).SetConfig(&data.FieldConfig{Unit: "m/s"}),
+		data.NewField("quality", nil, []string{"GOOD"}),
+	)
+	if diff := cmp.Diff(expectedFrame, qdr.Responses["A"].Frames[0], data.FrameTestCompareOptions()...); diff != "" {
+		t.Errorf("Result mismatch (-want +got):\n%s", diff)
 	}
+
+	mockSw.AssertExpectations(t)
```

Comment:
sorry for the very basic questions on my part but like what does this assert?

Reply:
It makes sure that all the all the declared methods ("mockSw.On("...")") were called.

Sometimes we declare those mocks but the test doesn't even use them which can be misleading. It's kind of an indirect way to assert some paths/coverage, I guess. 
I think some people might put these in everywhere, but I'm not sure it's always worth it. The reason I suggested it in these tests is that they're kind of "end to end," very long, complex, repetitive, and there are many to re-do. While pairing there were a few times we noticed that a test was written incorrectly and not testing what we thought.